app/helpers/dashboard_helper.rb

```diff
@@ -159,6 +159,20 @@ def out_of_sync_enabled?(origin)
     setting.nil? ? true : !setting
   end
 
+  def host_build_status_icon(host)
+    if host.token_expired?
+      icon = 'error-circle-o'
+      label = _('Token expired')
+    elsif host.build_errors.present?
+      icon = 'error-circle-o'
+      label = _('Build errors')
```

Comment:
There is one problem here - if a host has `build_errors` set, it is never in `build` mode. When I error a host by running this command:

    wget -q -O- --body-file=LICENSE --method POST --header="Content-Type: text/plain" --no-check-certificate http://localhost:3000/unattended/failed?token=689ba68a-a511-44b1-bca5-c8f3f1ff353e

then the host disappears. Therefore this branch can't be reached.

Reply:
Interestingly, I get this:

```
Encoding::UndefinedConversionError ("\xE2" from ASCII-8BIT to UTF-8):
```

Maybe we need something like this:

```ruby
.encode('utf-8', invalid: :replace, undef: :replace, replace: '_')
```